### PR TITLE
Remove `exit: return` idiom from src/inet and src/transport

### PR DESCRIPTION
**Pull request by kpschoedel**
#### Problem

The `Exit` group of error-handling macros inhibit scoping, and
are unnecessary when there is no cleanup at the `exit:` label.

#### Summary of Changes

- Remove `exit: return` idiom from src/lib, replacing `Exit` macros
  with corresponding `Return` macros.
- Incidentally fixed a bug in `RawEndPoint::BindInterface` and
  `UDPEndPoint::BindInterface` that could leave the LwIP lock held
  by calling `SuccessOrExit()` within a locked section.

### CODE REVIEW

**Comment by kpschoedel** [comment]
It seems that the unexpected code size growth in the previous version of this PR (#5914) was a CI fluke, probably a race in picking the master branch artifact. In fact ARM wins just a few bytes and Xtensa loses just a few bytes (in BLE::Init, for no obvious reason).

**Review by msandstedt — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: src/protocols/secure_channel/NetworkProvisioning.cpp [hunk 1]
```diff
@@ -156,38 +156,32 @@ size_t NetworkProvisioning::EncodedStringSize(const char * str)
 
 CHIP_ERROR NetworkProvisioning::EncodeString(const char * str, Encoding::LittleEndian::BufferWriter & bbuf)
 {
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-    size_t length   = strlen(str);
-    uint16_t u16len = static_cast<uint16_t>(length);
-    VerifyOrExit(CanCastTo<uint16_t>(length), err = CHIP_ERROR_INVALID_ARGUMENT);
+    const size_t length   = strlen(str);
+    const uint16_t u16len = static_cast<uint16_t>(length);
+    VerifyOrReturnError(CanCastTo<uint16_t>(length), CHIP_ERROR_INVALID_ARGUMENT);
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/NetworkProvisioning.cpp:161]: The order of these two lines should be switched, now that we can do it without the compiler complaining at us.  So do the check first, then the line with the cast.
```suggestion
    VerifyOrReturnError(CanCastTo<uint16_t>(length), CHIP_ERROR_INVALID_ARGUMENT);
    const uint16_t u16len = static_cast<uint16_t>(length);
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/protocols/secure_channel/NetworkProvisioning.cpp [hunk 2]
```diff
@@ -156,38 +156,32 @@ size_t NetworkProvisioning::EncodedStringSize(const char * str)
 
 CHIP_ERROR NetworkProvisioning::EncodeString(const char * str, Encoding::LittleEndian::BufferWriter & bbuf)
 {
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-    size_t length   = strlen(str);
-    uint16_t u16len = static_cast<uint16_t>(length);
-    VerifyOrExit(CanCastTo<uint16_t>(length), err = CHIP_ERROR_INVALID_ARGUMENT);
+    const size_t length   = strlen(str);
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/NetworkProvisioning.cpp:159]:
```suggestion
    const size_t length  = strlen(str);
```

File: src/protocols/secure_channel/NetworkProvisioning.cpp [hunk 3]
```diff
@@ -156,38 +156,32 @@ size_t NetworkProvisioning::EncodedStringSize(const char * str)
 
 CHIP_ERROR NetworkProvisioning::EncodeString(const char * str, Encoding::LittleEndian::BufferWriter & bbuf)
 {
-    CHIP_ERROR err  = CHIP_NO_ERROR;
-    size_t length   = strlen(str);
-    uint16_t u16len = static_cast<uint16_t>(length);
-    VerifyOrExit(CanCastTo<uint16_t>(length), err = CHIP_ERROR_INVALID_ARGUMENT);
+    const size_t length  = strlen(str);
```
Comment (bzbarsky-apple) [inline src/protocols/secure_channel/NetworkProvisioning.cpp:159]:
```suggestion
    const size_t length = strlen(str);
```

**Pull request merged**